### DISCUSSION TITLE
是否能进行裁剪发布

### DISCUSSION THREAD

**Discussion by NLick47** [yaobiao131/downkyicore#192] [General]
50mb的压缩包在github上还是太过于大了，我这里开启发布裁剪后压缩包能在30mb以下

**Comment by yaobiao131** [comment]
裁剪要测试各个功能是否正常，我本来不想带dotnet库的，但是防止不同dotnet版本兼容性问题带上了。因为用了一个dotnet7开始弃用的功能，后续准备修改一下兼容、你可以发出来测试一下

**Comment by NLick47** [comment]
```
<linker>
	<assembly fullname="DryIoc">
		<type fullname="*" />
	</assembly>

	<assembly fullname="Prism.DryIoc.Avalonia">
		<type fullname="*" />
	</assembly>


	<assembly fullname="Prism">
		<type fullname="*" />
	</assembly>


	<assembly fullname="Prism.Avalonia">
		<type fullname="*" />
	</assembly>


	<assembly fullname="Avalonia.Controls">
		<type fullname="*" />
	</assembly>

	<assembly fullname="Avalonia.Controls.DataGrid">
		<type fullname="*" />
	</assembly>


	<assembly fullname="DownKyi.Core">
		<type fullname="*" />
	</assembly>
	<assembly fullname="DownKyi">
		<type fullname="*" />
	</assembly>
	

</linker>
```
这是我的配置，我使用的.net8.0发布。可能某些序列化的地方我并没有修改正确

**Comment by NLick47** [comment]
[Uploading osx-64.zip…]()
压缩包没包含下载器和ffmpeg

**Comment by NLick47** [comment]
[win64.zip](https://github.com/user-attachments/files/18175238/win64.zip)

**Comment by NLick47** [comment]
[osx-64.zip](https://github.com/user-attachments/files/18175258/osx-64.zip)
2

**Comment by NLick47** [comment]
压缩包体积能在20mb以下，往这方面去做一点更新或许会更好

**Comment by NLick47** [comment]
作者你好，我最近又开始想对裁剪版进行测试和修改了，如果我这大部分功能测试通过以后windows的发行版可以再把裁剪版本也附带上吗，我也会对裁剪版本做一定维护

**Reply by yaobiao131** [reply]
可以的，你测试后可以提交pr，我还要测试下其他系统的兼容性

**Comment by NLick47** [comment]
我发现裁剪版本出现了如System.IO.Compression.ZipFile,System.IO.FileSystem.Watcher,System.Net.Ping等等不太应该会出现的程序集。而且还有Microsoft.VisualBasic.dll,Microsoft.VisualBasic.Core.dll很是疑惑

**Reply by NLick47** [reply]
感觉应该是ci的原因

**Reply by yaobiao131** [reply]
> 感觉应该是ci的原因

我在本地发布也存在，具体原因可能还需要排查一下

**Comment by NLick47** [comment]
[publish.zip](https://github.com/user-attachments/files/21749902/publish.zip)
这是我用裁剪配置在本地发布出来的可以看下，我在win-x64目录下看到有这些程序集，win-x64目录下的publish却没有。不知道是不是这个原因

**Reply by yaobiao131** [reply]
> [publish.zip](https://github.com/user-attachments/files/21749902/publish.zip) 这是我用裁剪配置在本地发布出来的可以看下，我在win-x64目录下看到有这些程序集，win-x64目录下的publish却没有。不知道是不是这个原因

已确认是ci pupnet打包工具的问题，下一版本我会修复这个问题

**Comment by NLick47** [comment]
Microsoft.DiaSymReader.Native.amd64.dll  这个程序集也需要被移除
